configure.ac

```diff
@@ -102,11 +109,29 @@ AS_VAR_IF([$1], [""], [$5], [$4])dnl
 ########################
 # Check for the dl library (specifically, the "dlsym" function)
 AC_CHECK_LIB([dl], [dlsym])
-# Check for UCX v1.6.0 or higher (Required)
-# TODO: make UCX optional based on a new AC_ARG_ENABLE flag
+# Find and get info for Flux-Core using pkg-config
+AX_FLUX_CORE
+# Check for Jansson 2.10 or higher (Required)
+PKG_CHECK_MODULES([JANSSON],
+    [jansson >= 2.10]
+)
+# Check for PerfFlow Aspect
+AX_PERFFLOW_ASPECT([PERFFLOW],
+    [pkg_check_perfflow_found=yes],
+    [pkg_check_perfflow_found=no]
+)
+if test "x$with_perfflow" = "xyes" && test "x$pkg_check_perfflow_found" = "xno"; then
+    AC_MSG_ERROR([requested PerfFlow Aspect support, but cannot find PerfFlow Aspect with pkg-config])
+fi
+# Check for UCX v1.6.0 or higher
 PKG_CHECK_MODULES([UCX],
-    [ucx >= 1.6.0]
+    [ucx >= 1.6.0],
+    [pkg_check_ucx_found=yes],
+    [pkg_check_ucx_found=no]
 )
+if test "x$with_ucx" = "xyes" && test "x$pkg_check_ucx_found" = "xno"; then
```

Comment:
should this be with_ucx or with enable-ucx? with-ucx is generally optional only to add when we don't know the path to its pkgconfig. I think enable-ucx should be the main check.

Reply:
I chose with-ucx because, based on the GNU docs, with-<pkg> is used when dealing with optional dependencies. enable-<feature> is used to enable features.

Reply:
To me ucx is a feature we added right?